net_transport.go

```diff
@@ -247,13 +247,21 @@ func (n *NetworkTransport) getPooledConn(target ServerAddress) *netConn {
 
 // getConnFromAddressProvider returns a connection from the server address provider if available, or defaults to a connection using the target server address
 func (n *NetworkTransport) getConnFromAddressProvider(id ServerID, target ServerAddress) (*netConn, error) {
+	address := n.getProviderAddressOrFallback(id, target)
+	return n.getConn(address)
+}
+
+func (n *NetworkTransport) getProviderAddressOrFallback(id ServerID, target ServerAddress) ServerAddress {
 	if n.serverAddressProvider != nil {
-		serverAddressOverride := n.serverAddressProvider.ServerAddr(id)
-		if serverAddressOverride != "" {
-			return n.getConn(serverAddressOverride)
+		serverAddressOverride, err := n.serverAddressProvider.ServerAddr(id)
+		if err != nil {
+			n.logger.Printf("[WARN] Unable to get address for server id %v, got error:%v", id, err)
```

Comment:
I'd make this just one log line since it's the same event (and make it a WARN for the one line).